controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -404,6 +423,127 @@ func (ew *EndpointsWatcher) getServicePublisher(id ServiceID) (sp *servicePublis
 	return
 }
 
+func (ew *EndpointsWatcher) addServer(obj interface{}) {
+	server := obj.(*v1beta1.Server)
+
+	// Get the set of pods that the server selects so that we can check if a
+	// port publisher exists for that endpoint, and if so it's new set of
+	// opaque pod ports is received.
+	options, err := createListOptions(server.Spec.PodSelector)
+	if err != nil {
+		ew.log.Errorf("failed to create Server ListOptions: %s", err)
+		return
+	}
+	pods, err := ew.k8sAPI.Client.CoreV1().Pods("").List(context.Background(), options)
```

Comment:
I don't think that it's necessary to get all pods here.

I think we can instead just iterate through the publishers and for each publisher, it can iterate through its address and check if any of them are pods whose labels match the server's selector.